SwiftWisdom/Core/UIKit/NibLoadableView.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  NibLoadableView.swift
+//  SwiftWisdom
+//
+//  Created by Christian Howland on 3/31/17.
+//  Copyright © 2017 Intrepid. All rights reserved.
+//
+
+import UIKit
+
+/**
+ Subclassing NibLoadableView allows you to automate the loading of a Nib into a placeholder view inside
+ another Nib, using only the Custom Class attribute.
+ 
+ It removes the need to perform the following manual steps every time you want a Nib that can load into a placeholder View:
+
+ - Setting an outlet for the top level view
+ - Setting 'translatesAutoresizingMaskIntoConstraints' to false
+ - Adding constraints to fit the new view into its container
+ - Writing boilerplate Nib loading code.
+
+ The minimum amount of code required to use this would just be the inheritance line, i.e.:
+
+ " YourCustomView: NibLoadableView {} "
```

Comment:
I understand the motivation. What I'm suggesting is a small twist on what you've built. 

Instead of recommending that NibLoadableView be inherited from, we should mark that class as _final_. Instead, expose a `IBInspectable` which allows one to enter the class name - then use your code as before, but now inspecting that property instead to figure out which class to load. This way, we don't have the custom type twice.

The reason being that custom type(YourCustomView) could be doing lots of things, listening to pub sub notifications, adding gesture recognizers, etc and it might be surprising to have both of those around.

In this respect, this class can maybe be called NibContainerView, or something similar. Lmk if that makes sense.